File: TauResearch/Views/Upload/UploadSection.swift
// UploadSection.swift
// Mirrors upload/ui/UploadSection.kt — embedded "Upload video" panel.

import SwiftUI

struct UploadSection: View {
    @Bindable var viewModel: VideoUploadViewModel
    let videoPath: String
    let gpxPath: String

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Button("Upload Video") {
                viewModel.startUpload(videoPath: videoPath, gpxPath: gpxPath)
            }
            .disabled(viewModel.state.isUploading)
            .buttonStyle(.borderedProminent)
            .tint(Color.brandGreen)

            if viewModel.state.isUploading {
                ProgressView(value: viewModel.state.progress)
                Button("Cancel Upload") { viewModel.cancelUpload() }
                    .buttonStyle(.bordered)
            }
            if let err = viewModel.state.error {
                Text(err).foregroundStyle(.red)
            }
            if viewModel.state.success {
                Label("Upload successful", systemImage: "checkmark.circle.fill")
                    .foregroundStyle(Color.brandGreen)
            }
        }
    }
}

File: app/src/main/java/com/example/cameraaccess/upload/ui/UploadSection.kt
package com.example.cameraaccess.upload.ui

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.material3.Button
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.example.cameraaccess.upload.viewmodel.VideoUploadViewModel

@Composable
fun UploadSection(
    viewModel: VideoUploadViewModel,
    videoPath: String,
    gpxPath: String
) {
    val state by viewModel.uploadState.collectAsState()

    Column {
        Button(
            onClick = { viewModel.startUpload(videoPath, gpxPath) },
            enabled = !state.isUploading
        ) {
            Text("Upload Video")
        }

        if (state.isUploading) {
            LinearProgressIndicator(
                progress = state.progress,
                modifier = Modifier.fillMaxWidth()
            )
            Spacer(Modifier.height(8.dp))
            Button(onClick = { viewModel.cancelUpload() }) {
                Text("Cancel Upload")
            }
        }

        if (state.error != null) {
            state.error?.let {
                Text(
                    text = it,
                    color = MaterialTheme.colorScheme.error
                )
            }
        }

        if (state.success) {
            Text("Upload successful ✅")
        }
    }
}